File: app/src/main/java/com/example/TibiaTools/View/ViewModel/ViewModelGuildInformation.java
package com.example.TibiaTools.View.ViewModel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.TibiaTools.APISERVER.models.GuildInformation.GuildName.Guild;
import com.example.TibiaTools.Repository.RepositoryGuildsInformation;

public class ViewModelGuildInformation extends ViewModel {
    RepositoryGuildsInformation repositoryGuildsInformation = new RepositoryGuildsInformation();
    private final MutableLiveData<Guild> _guild = new MutableLiveData<>();
    public MutableLiveData<Guild> guild(){ return _guild; }

    public void setGuild(String guildName){
        repositoryGuildsInformation.guildInformation(guildName, _guild);
    }
}

File: app/src/main/java/com/example/TibiaTools/View/ViewModel/ViewModelGuildInformation.kt
package com.example.TibiaTools.View.ViewModel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.TibiaTools.data.model.Guild
import com.example.TibiaTools.domain.usecase.GetGuildsInformationNameUseCase
import kotlinx.coroutines.launch

class ViewModelGuildInformation(
    private val getGuildsInformationNameUseCase: GetGuildsInformationNameUseCase
) : ViewModel() {
    private val _guild = MutableLiveData<Guild?>()
    fun guild(): LiveData<Guild?> = _guild

    fun setGuild(guildName: String) {
        viewModelScope.launch {
            runCatching { getGuildsInformationNameUseCase(guildName) }
                .onSuccess { _guild.value = it.guild }
                .onFailure { _guild.value = null }
        }
    }
}
